pkg/kubelet/util/pluginwatcher/plugin_watcher.go

```diff
@@ -158,10 +160,44 @@ func (w *Watcher) invokeRegistrationCallbackAtHandler(ctx context.Context, clien
 	}); err != nil {
 		return fmt.Errorf("failed to send registration status at socket %s, err: %v", socketPath, err)
 	}
+
 	chanForAckOfNotification <- true
 	return nil
 }
 
+// Handle filesystem notify event.
+func (w *Watcher) handlefsnotifyEvent(event fsnotify.Event) error {
+	if event.Op&fsnotify.Create != fsnotify.Create {
+		return nil
+	}
+
+	fi, err := os.Stat(event.Name)
+	if err != nil {
+		return fmt.Errorf("stat file %s failed: %v", event.Name, err)
+	}
+
+	if !fi.IsDir() {
+		err := w.registerPlugin(event.Name)
+		if err != nil {
+			return fmt.Errorf("Plugin %s registration failed with error: %v", event.Name, err)
+		}
```

Comment:
Hi @figo,

For the sake of readable code, please consider adding a `return` statement after this `if` block and then unindenting the `else` block below. For example:

```go
	if !fi.IsDir() {
		if err := w.registerPlugin(event.Name); err != nil {
			return fmt.Errorf("Plugin %s registration failed with error: %v", event.Name, err)
		}
		return nil
	}
	if filepath.Base(event.Name)[0] == '.' {
		return nil
	}
	if err := w.fsWatcher.Add(event.Name); err != nil {
		return fmt.Errorf("failed to watch %s, err: %v", event.Name, err)
	}
	if err := w.traversePluginDir(event.Name); err != nil {
		return fmt.Errorf("failed to traverse plugin path %s, err: %v", event.Name, err)
	}
	return nil
```